lib/iris/fileformats/grib/_save_rules.py

```diff
@@ -257,8 +257,18 @@ def latlon_first_last(x_coord, y_coord, grib):
 def dx_dy(x_coord, y_coord, grib):
     x_step = regular_step(x_coord)
     y_step = regular_step(y_coord)
-    gribapi.grib_set(grib, "DxInDegrees", float(abs(x_step)))
-    gribapi.grib_set(grib, "DyInDegrees", float(abs(y_step)))
+    if x_coord.units == 'degrees':
+        gribapi.grib_set(grib, "iDirectionIncrement",
+                         round(1e6 * float(abs(x_step))))
```

Comment:
the 1e6 is a multiplier to provide values in degrees x 10 ^ -6, as per the WMO manual on codes

the `round` ensures proper rounding to convert to an integer

this is to ensure that the `iDirectionIncrement` is set correctly.  this is a coded key, which are preferred by design in the iris implementation, unlike `DyInDegrees` which is computed.  the eccodes implementation has a subtle bug in its implementation, with different behaviour between py2 and py3

we ought to follow up on this with ECMWF, but I have not yet done this